applications/unit_tests/llri.cpp

```diff
@@ -9,6 +9,55 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
 #include <doctest/doctest.h>
 
+#include <helpers.hpp>
+
+TEST_CASE("print info")
+{
+    printf("LLRI unit tests\n");
+    llri::Instance* instance;
+    const llri::instance_desc desc{ 0, nullptr, "unit test instance"};
+    llri::createInstance(desc, &instance);
+    
+    std::vector<llri::Adapter*> adapters;
+    instance->enumerateAdapters(&adapters);
+    
+    printf("Available adapters: (%u)\n", static_cast<uint32_t>(adapters.size()));
+    for (auto* adapter : adapters)
+    {
+        llri::adapter_info info = adapter->queryInfo();
+        printf("\t\"%s\"\n", info.adapterName.c_str());
+    }
+    
+    if (strcmp(LLRI_SELECTED_TEST_ADAPTERS, "All") == 0)
+    {
+        printf("LLRI_SELECTED_TEST_ADAPTERS was set to \"All\", all adapters available are selected for unit test execution.\n");
+        printf("This configuration is %s\n", adapters.size() > 0 ? "valid" : "invalid");
+    }
+    else
+    {
+        printf("LLRI_SELECTED_TEST_ADAPTERS was set to %s\n", LLRI_SELECTED_TEST_ADAPTERS);
+        printf("Unit tests will attempt to run with the defined set of adapters. All adapters must be available for the unit tests to succeed.\n");
+        
+        auto selectedAdapterNames = detail::split(LLRI_SELECTED_TEST_ADAPTERS, ';');
+        std::vector<llri::Adapter*> selectedAdapters;
+
+        for (auto* adapter : adapters)
+        {
+            llri::adapter_info info = adapter->queryInfo();
+            
+            if (std::find(selectedAdapterNames.begin(), selectedAdapterNames.end(), info.adapterName) != selectedAdapterNames.end())
+            {
```

Comment:
same as mentioned before

Reply:
This part only runs once as an "info" pass - it's a little temporary but for now it's a way for unit test setups to confirm the validity of their configuration, I'm not worried about perf in this particular one